giving-example/index.js

```diff
@@ -0,0 +1,335 @@
+    // Conditionally handle different result codes for the shopper
+    switch (response.resultCode) {
+      case "Authorised":
+        res.redirect("/result/success");
+        break;
+      case "Pending":
+      case "Received":
+        res.redirect("/result/pending");
+        break;
+      case "Refused":
+        res.redirect("/result/failed");
+        break;
+      default:
+        res.redirect("/result/error");
+        break;
+    }
+  } catch (err) {
+    console.error(`Error: ${err.message}, error code: ${err.errorCode}`);
+    res.redirect("/result/error");
+  }
+});
+
+/* ################# end API ENDPOINTS ###################### */
+
+/* ################# CLIENT SIDE ENDPOINTS ###################### */
+
+// Index (select a demo)
+app.get("/", (req, res) => res.render("index"));
+
+// Cart (continue to checkout)
+app.get("/preview", (req, res) =>
+  res.render("preview", {
+    type: req.query.type,
+  })
+);
+
+// Checkout page (make a payment)
+app.get("/checkout", (req, res) =>
+  res.render("checkout", {
+    type: req.query.type,
+    clientKey: process.env.ADYEN_CLIENT_KEY,
+  })
+);
+
+// Result page
+app.get("/result/:type", (req, res) =>
+  res.render("result", {
+    type: req.params.type,
+    clientKey: process.env.ADYEN_CLIENT_KEY,
+  })
+);
+
+/* ################# end CLIENT SIDE ENDPOINTS ###################### */
+
+/* ################# WEBHOOK ###################### */
+
+// Process incoming Webhook: get NotificationRequestItem, validate HMAC signature,
+// consume the event asynchronously, send response ["accepted"]
+app.post("/api/webhooks/notifications", async (req, res) => {
+  console.log("/api/webhooks/notifications");
+
+  // YOUR_HMAC_KEY from the Customer Area
+  const hmacKey = process.env.ADYEN_HMAC_KEY;
+  const validator = new hmacValidator()
+  // Notification Request JSON
+  const notificationRequest = req.body;
+  const notificationRequestItems = notificationRequest.notificationItems
+
+  // fetch first (and only) NotificationRequestItem
+  const notification = notificationRequestItems[0].NotificationRequestItem
+  console.log(notification)
+  
+  // Handle the notification
+  if( validator.validateHMAC(notification, hmacKey) ) {
+    // valid hmac: process event
+    const merchantReference = notification.merchantReference;
+    const eventCode = notification.eventCode;
+    console.log("merchantReference:" + merchantReference + " eventCode:" + eventCode);
+
+    // consume event asynchronously
+    consumeEvent(notification);
+
+    // acknowledge event has been consumed
+    res.send('[accepted]')
+
+  } else {
+    // invalid hmac: do not send [accepted] response
+    console.log("Invalid HMAC signature: " + notification);
+    res.status(401).send('Invalid HMAC signature');
+  }
+
+});
+
+// Process incoming Giving Webhook: get NotificationRequestItem,
+// consume the event asynchronously, send response ["accepted"]
+app.post("/api/webhooks/giving", async (req, res) => {
```

Comment:
Any reason why we have to separate these two endpoints? (And why we're not validating the HMAC of this webhook?)

Reply:
2 different webooks in CA, Giving doesnt support HMAC validation